sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformProvider.java

```diff
@@ -178,16 +193,45 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
       if (confluentSchemaRegUrl != null) {
         final String confluentSchemaRegSubject =
             checkArgumentNotNull(configuration.getConfluentSchemaRegistrySubject());
-        KafkaIO.Read<byte[], GenericRecord> kafkaRead =
+        KafkaIO.Read<byte[], GenericRecord> kafkaRead;
+
+        kafkaRead =
             KafkaIO.<byte[], GenericRecord>read()
                 .withTopic(configuration.getTopic())
                 .withConsumerFactoryFn(new ConsumerFactoryWithGcsTrustStores())
                 .withBootstrapServers(configuration.getBootstrapServers())
                 .withConsumerConfigUpdates(consumerConfigs)
-                .withKeyDeserializer(ByteArrayDeserializer.class)
-                .withValueDeserializer(
+                .withKeyDeserializer(ByteArrayDeserializer.class);
+
+        SchemaRegistryProvider provider = getSchemaRegistryProvider(confluentSchemaRegUrl);
+        switch (provider) {
+          case GOOGLE_MANAGED:
+            // Custom configs to authenticate with Google's Managed Schema Registry
+            Map<String, Object> configs = new HashMap<>();
+            configs.put(
+                KafkaAvroDeserializerConfig.SCHEMA_REGISTRY_URL_CONFIG, confluentSchemaRegUrl);
+            configs.put(KafkaAvroDeserializerConfig.BEARER_AUTH_CREDENTIALS_SOURCE, "CUSTOM");
+            configs.put(
+                "bearer.auth.custom.provider.class",
+                "com.google.cloud.hosted.kafka.auth.GcpBearerAuthCredentialProvider");
+
+            LOG.info(
```

Comment:
Will the full url contain sensitive field or PII (e.g. user name, token, etc)? If so we'd better do not log the full url